### ISSUE TITLE
connection aborted when stopped while loading

### ISSUE THREAD

**Issue by immadisairaj** [ryanheise/just_audio#374]
**Which API doesn't behave as documented, and how does it misbehave?**
When the player starts loading, I try to stop it (because I don't want to hear that). When I try to stop it while it's still loading in the background, I get the error.

**Minimal reproduction project**
```dart
  // handle the pause and play button
  void _handleOnPressed(bool isPlaying) {
    if(!isPlaying) {
      _player.setAudioSource(AudioSource.uri(
        Uri.parse('$audioSource')
        ), preload: false);
      await _player.load();
      await _player.play();
    } else {
      await _play.stop();
    }
  }
```

**To Reproduce (i.e. user steps, not code)**
1. Press the Play Button
2. Press the stop button immediately while loading
3. It gives out the exception and my loading indicator doesn't stop (it updates based on the processing states)

**Error messages**

```
I/ExoPlayerImpl(30222): Release a7f0f5e [ExoPlayerLib/2.13.1] [RMX1921, RMX1921, realme, 29] [goog.exo.core]
E/flutter (30222): [ERROR:flutter/lib/ui/ui_dart_state.cc(186)] Unhandled Exception: Connection aborted
E/flutter (30222): #0      AudioPlayer._load (package:just_audio/just_audio.dart:711:11)
E/flutter (30222): <asynchronous suspension>
E/flutter (30222): #1      AudioPlayer._setPlatformActive.<anonymous closure> (package:just_audio/just_audio.dart:1086:28)
E/flutter (30222): <asynchronous suspension>
E/flutter (30222): #2      AudioPlayer._setPlatformActive.<anonymous closure> (package:just_audio/just_audio.dart:1008:29)
E/flutter (30222): <asynchronous suspension>
E/flutter (30222):
```

**Expected behavior**
If the stop button is pressed while loading, the service has to stop and be ready for the next play service

**Screenshots**
The problem is not in the UI (except for the loading indicator doesn't stop)

**Desktop (please complete the following information):**
 - OS: Windows 10 Version 2004 (KB5000802)
 - Browser: maybe not required

**Smartphone (please complete the following information):**
 - Device: realmeXT
 - OS: Android 10

**Flutter SDK version**
```
[✓] Flutter (Channel beta, 2.1.0-12.2.pre, on Microsoft Windows [Version 10.0.19041.867], locale en-IN)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
[✗] Chrome - develop for the web (Cannot find Chrome executable at .\Google\Chrome\Application\chrome.exe)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.
[✓] Android Studio (version 4.0)
[✓] VS Code (version 1.55.1)
[✓] Connected device (1 available)
```

**Additional context**
I saw a recent issue that got closed already (due to no proper format) #359. I think this may be related to the same issue?
I also saw an answer from stack overflow [link](https://stackoverflow.com/questions/66580270/issue-in-just-audio-pugin-in-flutter). Maybe this is also related to the same issue? (PS: that answer didn't work for me)

package version: just_audio: ^0.7.3
I combined this using audio_service (which doesn't seem to be an issue for this problem).
PS: due to this issue, I tried to block stopping the service while it is loading (as a workaround).

**Comment by ryanheise** [comment]
This is actually the intended behaviour according to this section in the README:

Catching player errors: 

```dart
try {
  await player.setUrl("https://s3.amazonaws.com/404-file.mp3");
} on PlayerException catch (e) {
  // iOS/macOS: maps to NSError.code
  // Android: maps to ExoPlayerException.type
  // Web: maps to MediaError.code
  print("Error code: ${e.code}");
  // iOS/macOS: maps to NSError.localizedDescription
  // Android: maps to ExoPlaybackException.getMessage()
  // Web: a generic message
  print("Error message: ${e.message}");
} on PlayerInterruptedException catch (e) {
  // This call was interrupted since another audio source was loaded or the
  // player was stopped or disposed before this audio source could complete
  // loading.
  print("Connection aborted: ${e.message}");
} catch (e) {
  // Fallback for all errors
  print(e);
}
```

It will throw the exception to indicate what happened and you can choose to catch it and ignore it if your app doesn't need to do anything special in this case.

**Comment by immadisairaj** [comment]
Thanks for your response @ryanheise. I got to know the issue and tried to solve it.

The issue was with set/loading the resource when I am trying to play for the second time.

Before (from the code I gave above):
1. Tried to load the resource
2. Play
3. Stop it while loading
4. Play again (here it plays the audio before setting the resource)

After:
I forgot to add the "await" while setting the source. So, after looking through the code you sent and the README (again) I changed it to the below code.
```dart
 // handle the pause and play button
  void _handleOnPressed(bool isPlaying) {
    if(!isPlaying) {
      _player.play();
      await _player.setAudioSource(AudioSource.uri(
        Uri.parse('$audioSource')
        ));
    } else {
      await _play.stop();
    }
  }
```
Now, it works the way I want. Sorry for the trouble and thanks for helping out.

I guess, this issue can be closed.
